lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -361,6 +370,35 @@ def run
     end
   end
 
+  def ipv4_address(vm)
+    puts 'Waiting for network interfaces to become available...'
+    sleep 2 while vm.guest.net.empty?
+    puts 'Waiting for a valid IPv4 address...'
+    # Multiple reboots occur during guest customization in which a link-local
+    # address is assigned. As such, we need to wait until a routable IP address
+    # becomes available. This is most commonly an issue with Windows instances.
+    sleep 2 while vm.guest.net[config[:bootstrap_nic]].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer'
+    vm.guest.net[config[:bootstrap_nic]].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.ipAddress
+  end
+
+  def guest_address(vm)
+    guest_address ||=
+      config[:fqdn] = if config[:bootstrap_ipv4]
+                        ipv4_address(vm)
+                      elsif config[:fqdn]
+                        get_config(:fqdn)
+                        sleep 2 until vm.guest.ipAddress
```

Comment:
That makes sense. All 3 branches of this `if` statement now have some kind of `sleep` (`ipv4_address` calls it at the top), do you think we could pull it out to the top of `guest_address` so that it's more DRY, not to mention clear that it's important that we wait before continuing?